### DISCUSSION TITLE
Project Update and Tasks #3

### DISCUSSION THREAD

**Discussion by Joseph-Kassab** [DrAlzahraniProjects/csusb_fall2024_cse6550_team2#97] [Announcements]
Hello everyone,

**Our project is due November 25. So we need everyone helping. I made an action plan with Farheen today that I'll share.**

### **Priority tasks for next Tuesday's presentation (Q&A post on Canvas):**

1. Vishnu and Jason - Need a Jupyter notebook of our implementation with markdown comments and headers (talk to Farheen).
2. Arun and Jesse - Must show a confusion matrix of accuracy metrics in chatbot.
3. Sandhya and Sujana - Must have an SQA document of 20 questions (10 that chatbot gets correct, and 10 that it doesn't).

### **Individual priorities:**

**Prasanth**: 
Responsible for documenting data from the following websites:  https://www.csusb.edu/cse and https://catalog.csusb.edu/. Track what information (text, images, URLs) each page contains, including any nested pages.

**Arun**:
Lead the backend and UI development for the thumbs up/down feature and the 10 statistics report. Ensure the UI includes a live confusion matrix displaying six key metrics—Sensitivity, Specificity, Accuracy, Precision, Recall, and F1 Score—with a reset button below for clearing tests.

**Jesse**:
Assist Arun with UI development, including the integration of the live confusion matrix and feedback icons. Support Farheen with code testing and documentation of the Jupyter Notebook code.

**Shreya**:
Conduct a Docker Scout scan and produce a report in GitHub Discussions. Additionally, help Visnu and Jason with documenting the Jupyter Notebook's interactive mode demo, ensuring code is documented both inside and outside the cells, with output displayed after each execution.

**Shishir**:
Assist Farheen with Mamba-related issues and contribute to the documentation of the Jupyter Notebook as needed.

**Vishnu and Jason**:
Handle Jupyter Notebook for presentation and ensure web port 6002 is configured and accessible. Prepare an interactive mode demo showcasing the functionality of our chatbot with output displayed for each block of code (can be print statements).

**Sandhya**:
Oversee actions and discussions pages; all team members should contribute to discussions. Manage pull requests—coordinate merging with relevant team members or consult Farheen. Prepare the SQA document with 10 answerable and 10 unanswerable questions.

**Sujana**:
Assist Sandhya with maintaining GitHub pages and managing discussions. Help compile the SQA document alongside Sandhya.

**Vedakshari**:
Use dummy websites with vulnerabilities to implement ZAP and BSCE. Produce security reports using ZAP and BSCE for our project in GitHub discussions.

**Siyamul**:
Remove warnings and enhance performance in Docker. Assist Vedakshari with security implementations.

**Abhishek**:
Help Farheen test code across different platforms (Mac, Windows, Linux) using virtual machines. Provide support for testing the confusion matrix functionality.

**Jahnavi**:
Responsible for SRS, and collaborate with Farheen for code testing.

**Comment by farheen-akhter-code** [comment]
@ketha-shishir Pls look at the current docker file and see what all packages can be installed via mamba instead of using pip
